sleap_nn/data/providers.py

```diff
@@ -265,15 +266,129 @@ def run(self):
                 img = np.expand_dims(img, axis=0)  # (1, C, H, W)
 
                 self.frame_buffer.put(
-                    (
-                        torch.from_numpy(img),
-                        torch.tensor(idx, dtype=torch.int32),
-                        torch.Tensor(img.shape[-2:]),
-                    )
+                    {
+                        "image": torch.from_numpy(img),
+                        "frame_idx": torch.tensor(idx, dtype=torch.int32),
+                        "video_idx": torch.tensor(0, dtype=torch.int32),
+                        "orig_size": torch.Tensor(img.shape[-2:]),
+                    }
                 )
 
         except Exception as e:
             print(f"Error when reading video frame. Stopping video reader.\n{e}")
 
         finally:
-            self.frame_buffer.put((None, None, None))
+            self.frame_buffer.put(
+                {
+                    "image": None,
+                    "frame_idx": None,
+                    "video_idx": None,
+                    "orig_size": None,
+                }
+            )
+
+
+class LabelReader(Thread):
```

Comment:
```suggestion
class LabelsReader(Thread):
```

And rename the old one: `LabelsReaderDP` so we don't have to remove it right away.